server/modules/analysis/service/rChain/index.js

```diff
@@ -107,26 +105,18 @@ export const persistUserScripts = async ({ surveyId, chainUuid, filePath }) => {
       }
     })
 
-    const chainNodeDefs = Chain.getChainNodeDefs(chain)
-    const chainNodeDefsWithNodeDef = chainNodeDefs.map((chainNodeDef) => ({
-      ...chainNodeDef,
-      nodeDef: Survey.getNodeDefByUuid(chainNodeDef.node_def_uuid)(survey),
-    }))
-
     await PromiseUtils.each(entities, async (entity, entityIndex) => {
-      const chainNodeDefsInEntity = chainNodeDefsWithNodeDef.filter(
-        (chainNodeDef) => NodeDef.getParentUuid(chainNodeDef.nodeDef) === NodeDef.getUuid(entity)
-      )
+      const AnalysisNodeDefsInEntity = Survey.getAnalysisNodeDefs({ entity, chain })(survey)
```

Comment:
...the name of the constant should be lower case!

Reply:
hahahah too many refactor